### PR TITLE
Alphabetically sort the rules and enforce that

### PR DESCRIPTION
**Pull request by stroncium**
Added rules to enforce alphabetical sorting and rule presence in readme and index
added missing rules in readme
sorted rules

### CODE REVIEW

**Comment by stroncium** [comment]
integration test error inherited from master

File: test/package.js [hunk 1]
```diff
@@ -24,6 +34,48 @@ test('Every rule is defined in index file', t => {
 		'There are more exported rules than rule files.');
 	t.is(Object.keys(index.configs.recommended.rules).length, ruleFiles.length,
 		'There are more exported rules in the recommended config than rule files.');
+
+	testSorted(t, Object.keys(index.configs.recommended.rules), 'configs.recommended.rules');
```
Comment (futpib) [inline test/package.js]: Could also use [`sort-keys`](https://eslint.org/docs/rules/sort-keys) rule.

Comment (sindresorhus) [reply test/package.js]: Seems like overkill to setup ESLint to run programmatically just for this one thing.

**Comment by sindresorhus** [comment]
Perfect :)

File: test/package.js (hunk 1 above)
Comment (futpib) [reply test/package.js]: I meant to suggest `/* eslint-enable sort-keys */`

Comment (sindresorhus) [reply test/package.js]: Ah ok. Yeah, that would work.

**Pull request merged**